torch/serialization.py

```diff
@@ -353,6 +361,35 @@ def _meta_deserialize(obj, location):
     if location == 'meta':
         return torch.UntypedStorage(obj.nbytes(), device='meta')
 
+def validate_xpu_device(location):
+    xpu = getattr(torch, "xpu", None)
+    assert xpu is not None, "XPU device module is not loaded"
+    device = xpu._utils._get_device_index(location, optional=True)
+
+    if not xpu.is_available():
+        raise RuntimeError('Attempting to deserialize object on a XPU '
+                           'device but torch.xpu.is_available() is False. '
+                           'If you are running on a CPU-only machine, '
+                           'please use torch.load with map_location=torch.device(\'cpu\') '
+                           'to map your storages to the CPU.')
+    device_count = xpu.device_count()
+    if device >= device_count:
+        raise RuntimeError('Attempting to deserialize object on XPU device '
+                           f'{device} but torch.xpu.device_count() is {device_count}. Please use '
+                           'torch.load with map_location to map your storages '
+                           'to an existing device.')
+    return device
+
+def _xpu_deserialize(obj, location):
+    if location.startswith('xpu'):
+        xpu = getattr(torch, "xpu", None)
+        assert xpu is not None, "XPU device module is not loaded"
```

Comment:
`_cuda_deserialize`, `hpu_deserialize`, and `_xpu_deserialize` have the same logic. Please address @albanD  [comment](https://github.com/pytorch/pytorch/issues/125529#issuecomment-2096840784) to make the code more generic.

Reply:
Just want to note that `_privateuse1_deserialize` is another one with the same logic. Generalization will need to cover it as well.

Reply:
Ok, I generalized `_privateuse1_deserialize` to cover cuda, hpu, xpu and privateuse1 backends.